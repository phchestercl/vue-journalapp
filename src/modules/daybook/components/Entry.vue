<template>
    <div class="entry-container mb-3 pointer p-2"
        @click="$router.push({name:'entry', params:{id:10}}) "
    >
        <!-- titulo -->
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">15 </span>
            <span class="mx-1 fs-5">Septiembre </span>
            <span class="mx-1 fw-light">2021</span>
        </div>
        <div class="entry-description fs-6">
            Esse nostrud voluptate aliquip nostrud officia. Labore dolore occaecat minim aliquip laborum ea ea occaecat mollit et tempor anim sint. Minim minim sunt est aliqua commodo dolore ipsum labore Lorem ut anim ullamco elit culpa.
        </div>
    </div>
</template>
<style lang="scss" scoped>
.entry-container{
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;
    &:hover{
        background-color:lighten($color:rgb(99, 97, 97),$amount:45);
        transition: 0.2s all ease-in;
    }
    .entry-description{
        font-size: 12px;
    }
}
</style>



