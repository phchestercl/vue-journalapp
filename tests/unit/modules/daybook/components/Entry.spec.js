import { shallowMount } from '@vue/test-utils'
import Entry from '@/modules/daybook/components/Entry.vue'
import { journalState } from '../../../mock-data/test-journal-state'


//console.log(journalState.entries)

describe('Entry Component', () => {
    let mockRouter
    let wrapper

    beforeEach(() => {
        mockRouter = {
            push: jest.fn()
        }
        wrapper = shallowMount(Entry, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            },
            props: {
                entry: journalState.entries[0]
            }
        })

    })
    //mock Router
    // const router
    // Match con snapshot

    test('Match snapshot', () => {
        //console.log(journalState)
        //console.log(wrapper.html())
        expect(wrapper.html()).toMatchSnapshot()
    })

    // Debe redireccionar al hacer click en el entry container
    test('Redireccionar al hacer click', async () => {
        const entryContainer = wrapper.find('.entry-container')
        //console.log(entryContainer)
        await entryContainer.trigger('click')
        expect(mockRouter.push).toHaveBeenCalledTimes(1)
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'entry', params: { id: journalState.entries[0].id } })
    })
    // pruebas en las propiedades computadas
    // wrapper.vm <--- ver la propiedades computadas
    test('Pruebas en Propuedades computadas', () => {
        const wrapper = shallowMount(Entry, {
            /* global:{
                mocks:{
                    $router:mockRouter
                }
            }, */
            props: {
                entry: journalState.entries[0]
            }
        })
        //console.log(wrapper.vm.entry.date)
        //const yearDay = wrapper.vm.yearDay
        //console.log(yearDay)
        expect(wrapper.vm.day).toBe(7)
        expect(wrapper.vm.month).toBe('Septiembre')
        expect(wrapper.vm.yearDay).toBe('2021, Martes')
    })
})