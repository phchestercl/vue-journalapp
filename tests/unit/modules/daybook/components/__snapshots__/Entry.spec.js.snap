// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Component Match snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- titulo -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">7</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-1 fw-light">2021, Martes</span></div>
  <div class="entry-description fs-6">Carlos Mazu Mazda BT-50</div>
</div>
`;
